### PR TITLE
Use the same format for email styles as it is in theme.json

### PR DESCRIPTION
**Pull request by lysyjan**
## Description

_N/A_

## Code review notes

This PR is the second in a row related to the settings styles panel.
- The goal here is to prepare the style data structure for merging and allowing easier manipulation on the front end.

## QA notes

We can skip QA here and ask for testing in the last PR.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5640]

## After-merge notes

_N/A_

## Tasks

- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-5640]: https://mailpoet.atlassian.net/browse/MAILPOET-5640?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/assets/js/src/email-editor/engine/store/types.ts [hunk 1]
```diff
@@ -27,24 +27,31 @@ export type ExperimentalSettings = {
...
-      color: string;
-      fontFamily: string;
-      fontWeight: string;
+      color: {
+        text: string;
+      };
+      typography: {
+        fontFamily: string;
+        fontWeight: string;
+      };
```
Comment (costasovo) [inline mailpoet/assets/js/src/email-editor/engine/store/types.ts:54]: The change was added earlier, but I struggle to understand why we define only h1 in this type definition. Shouldn't we have some common type for element styles?

Comment (costasovo) [reply mailpoet/assets/js/src/email-editor/engine/store/types.ts:54]: Oh I see now h1 is used in `email-editor/engine/components/styles-sidebar/preview.tsx`.
Ok. I guess we can keep it for now. We are aiming to align with theme.json, so I'm hoping in the future, we will be able to grab some types from Gutenberg.

**Review by costasovo — DISMISSED** [review]
@lysyjan Thanks for unifying the format. It looks like it was a tedious job to find all the places. Great job!

File: mailpoet/assets/js/src/email-editor/engine/store/types.ts (hunk 1 above)
Comment (lysyjan) [reply mailpoet/assets/js/src/email-editor/engine/store/types.ts:54]: @costasovo I think we will need to add more definitions when we implement the typography panel. So, our theme.json will be more robust in the future.

**Review by costasovo — APPROVED** [review]

**Pull request merged**